documentation/List-of-Action-plugins.md

```diff
@@ -2,59 +2,256 @@
+
+- _Action Code:_ **`updateReviewAssignmentsStatus`**
+
+| Input parameters<br />(\*required) <br/>            | Output properties                                     |
+| --------------------------------------------------- | ----------------------------------------------------- |
+| `reviewAssignmentId`\*                              | `reviewAssignmentUpdates` (`array` of `{id, status}`) |
+| `trigger`\* (only executes on `onReviewSelfAssign`) |                                                       |
+
+---
+
+### Trim Responses
+
+Whenever an application or review is submitted, this Action "cleans up", by deleting all responses that have not changed since the last submission (or are `null`). Also, the remaining responses have their timestamps (`timeUpdated`) updated to match the timestamp of the most recent status_history (application or review, as appropriate).
+
+- _Action Code:_ **`trimResponses`**
+
+| Input parameters<br />(\*required) <br/> | Output properties |
+| ---------------------------------------- | ----------------- |
+| `applicationId` _OR_                     | `deletedIds`      |
+| `reviewId`                               | `updatedIds`      |
+
+---
+
+### Update Review Visibility
+
+Updates the applicant visibility of level 1 review responses based on the recommendations of the last-level review. I.e. it generate the "List of Questions" the applicant sees for making changes to their application.
+
+- _Action Code:_ **`updateReviewVisibility`**
+
+| Input parameters<br />(\*required) <br/> | Output properties                      |
+| ---------------------------------------- | -------------------------------------- |
+| `reviewId`\*                             | `reviewResponsesWithUpdatedVisibility` |
+
+---
+
+### Update Review Statuses
+
+When an applicant re-submits an application after making changes, this Action updates the status of associated reviews to determine whether they should be "Pending" or "Locked" (or left as is)
+
+- _Action Code:_ **`updateReviews`**
+
+| Input parameters<br />(\*required) <br/> | Output properties |
+| ---------------------------------------- | ----------------- |
+| `applicationId`\*                        | `updatedReviews`  |
+
+## Core Actions
+
+There are certain Actions that _must_ run on particular to facilitate a standard application/review workflow process. We have called these **Core Actions**, and they have been collected in a single "core_actions.js" file (for insertion into database via GraphQL). Each template (other than "User Registration") has this block slugged into it as a template literal (`${coreActions}`) in its Action insertion block, before any Actions that are specific to that template.
```

Comment:
`run on particular` I'm not familiar with this expression. Is this correct?

Reply:
Sorry, missing a word. Should be "run on particular events"

Reply:
Cool. Thanks : )